File: java/app/src/main/java/tech/cherri/tokenpushexample/fragment/TermsFragment.java
package tech.cherri.tokenpushexample.fragment;

import android.app.Fragment;
import android.app.FragmentManager;
import android.app.FragmentTransaction;
import android.os.AsyncTask;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.CheckBox;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import tech.cherri.tokenpushexample.MainActivity;
import tech.cherri.tokenpushexample.R;
import tech.cherri.tokenpushexample.TokenPushParam;
import tech.cherri.tokenpushexample.TokenPushTask;

public class TermsFragment extends Fragment {

    private CheckBox userAgreeCheckBox;
    private Button agreeBtn;
    private Button cancelBtn;
    private TokenPushParam tokenPushParam;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_add_card_terms, container, false);
    }

    @Nullable
    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        userAgreeCheckBox = view.findViewById(R.id.user_agree_checkbox);
        userAgreeCheckBox.setOnClickListener(v -> {
            agreeBtn.setEnabled(userAgreeCheckBox.isChecked());
        });
        cancelBtn = view.findViewById(R.id.cancel_button);
        cancelBtn.setOnClickListener(v -> {
            FragmentManager fragmentManager = getFragmentManager();
            FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
            fragmentTransaction.remove(TermsFragment.this);

            AddCardUserCancelFragment addCardUserCancelFragment = new AddCardUserCancelFragment();
            addCardUserCancelFragment.setTokenPushParam(tokenPushParam);
            fragmentTransaction.replace(R.id.fragment_container_view, addCardUserCancelFragment, AddCardUserCancelFragment.class.getSimpleName());
            fragmentTransaction.commit();
        });
        agreeBtn = view.findViewById(R.id.agree_button);
        agreeBtn.setOnClickListener(v -> {
            ((MainActivity) getActivity()).showProgressing();
            TokenPushTask tokenPushTask = new TokenPushTask(tokenPushParam);
            tokenPushTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);

            FragmentManager fragmentManager = getFragmentManager();
            FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
            fragmentTransaction.remove(TermsFragment.this);
            fragmentTransaction.commit();
        });
    }

    public TokenPushParam getTokenPushParam() {
        return tokenPushParam;
    }

    public void setTokenPushParam(TokenPushParam tokenPushParam) {
        this.tokenPushParam = tokenPushParam;
    }
}

File: kotlin/app/src/main/java/tech/cherri/tokenpushexample/fragment/TermsFragment.kt
package tech.cherri.tokenpushexample.fragment

import AddCardUserCancelFragment
import android.app.Fragment
import android.os.AsyncTask
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.CheckBox
import tech.cherri.tokenpushexample.MainActivity
import tech.cherri.tokenpushexample.R
import tech.cherri.tokenpushexample.TokenPushParam
import tech.cherri.tokenpushexample.TokenPushTask

class TermsFragment : Fragment() {
    private var tokenPushParam: TokenPushParam? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_add_card_terms, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        var userAgreeCheckBox: CheckBox = view.findViewById(R.id.user_agree_checkbox)
        var agreeBtn: Button = view.findViewById(R.id.agree_button)
        userAgreeCheckBox.setOnClickListener(View.OnClickListener { v: View? ->
            agreeBtn.isEnabled = userAgreeCheckBox!!.isChecked()
        })
        var cancelBtn: Button = view.findViewById(R.id.cancel_button)
        cancelBtn.setOnClickListener(View.OnClickListener { v: View? ->
            val fragmentManager = fragmentManager
            val fragmentTransaction = fragmentManager.beginTransaction()
            fragmentTransaction.remove(this@TermsFragment)
            val addCardUserCancelFragment = AddCardUserCancelFragment()
            addCardUserCancelFragment.setTokenPushParam(tokenPushParam)
            fragmentTransaction.replace(
                R.id.fragment_container_view,
                addCardUserCancelFragment,
                AddCardUserCancelFragment::class.java.getSimpleName()
            )
            fragmentTransaction.commit()
        })
        agreeBtn = view.findViewById(R.id.agree_button)
        agreeBtn.setOnClickListener(View.OnClickListener { v: View? ->
            (activity as MainActivity).showProgressing()
            val tokenPushTask = tokenPushParam?.let { TokenPushTask(it) }
            tokenPushTask?.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR)
            val fragmentManager = fragmentManager
            val fragmentTransaction = fragmentManager.beginTransaction()
            fragmentTransaction.remove(this@TermsFragment)
            fragmentTransaction.commit()
        })
    }

    fun getTokenPushParam(): TokenPushParam? {
        return tokenPushParam
    }

    fun setTokenPushParam(tokenPushParam: TokenPushParam?) {
        this.tokenPushParam = tokenPushParam
    }
}